### PR TITLE
Issue #207 - network rm (via API & UI) must fail for reserved networks

### PR DESCRIPTION
**Pull request by nerdalert**
-Fail network deletes for bridge, none and host.

-Gist w/ CLI validation: https://gist.github.com/f2667c3bd8a0392b447e

Signed-off-by: Brent Salisbury <brent.salisbury@docker.com>

### CODE REVIEW

File: api/api.go [hunk 1]
```diff
@@ -375,6 +375,9 @@ func procDeleteNetwork(c libnetwork.NetworkController, vars map[string]string, b
 	if !errRsp.isOK() {
 		return nil, errRsp
 	}
+	if isReservedNetwork(nw.Name()) {
```
Comment (aboch) [inline api/api.go]: It would be nice if checks like this one were managed by the backend in libnetwork. api pkg should not know about this, otherwise we are coupling the rest server with the backend.
But we do not have an API in libnetwork where to handle this check, given the delete network is handled directly by the network object. 
Actually, maybe we can put this logic in Network.Delete() and return a types.BadRequestf() if the net name is a reserved one.
Basically, the Network object itself will do the check on its name.
Can you please see if we can make the change that way. Thanks.

Comment (mrjana) [inline api/api.go]: @aboch @nerdalert In fact my suggestion is not even do it libnetwork but do it instead in the driver. And simply check with the driver before you delete a network object. For `null` and `host` it's easy. Always return an error for DeleteNetwork api in the driver. For `bridge` do not delete it if the `bridgeName` is `DefaultBridgeName`. And that's it. No special casing at all in the libnetwork code.

Comment (aboch) [inline api/api.go]: That's even better.

**Comment by mrjana** [comment]
@nerdalert can you please rework the code as per the comments?

**Comment by mavenugo** [comment]
Closing this in favor of : https://github.com/docker/libnetwork/pull/252
